pandas/tests/frame/test_reshape.py

```diff
@@ -140,6 +140,17 @@ def test_stack_mixed_level(self):
         expected = expected[["a", "b"]]
         tm.assert_frame_equal(result, expected)
 
+    def test_unstack_not_consolidated(self):
+        # Gh#34708
+        df = pd.DataFrame({"x": [1, 2, np.NaN], "y": [3.0, 4, np.NaN]})
+        df2 = df[["x"]]
```

Comment:
don't you need to copy here?

Reply:
no, we dont care about getting a copy, just non-consolidatedness, which is checked 2 lines down